### PR TITLE
fix coloring palette

### PR DESCRIPTION
**Pull request by Skalii**

### CODE REVIEW

File: Sources/Helpers/Appearance/OAGPXAppearanceCollection.mm [hunk 1]
```diff
@@ -453,6 +413,7 @@ - (OAColorItem *)getDefaultLineColorItem
         [_availableColors addObject:_defaultLineColorItem];
         _defaultLineColorItem.sortedPosition = [_availableColors indexOfObject:_defaultLineColorItem];
         [_defaultLineColorItem generateId];
+        _defaultColorValues[@"red"] = @(0xFFFF0000);
```
Comment (PaulStets) [inline Sources/Helpers/Appearance/OAGPXAppearanceCollection.mm:416]: Use color constant here!

File: Sources/Helpers/Appearance/OAGPXAppearanceCollection.mm [hunk 2]
```diff
@@ -461,10 +422,11 @@ - (OAColorItem *)getDefaultPointColorItem
 {
     if (!_defaultPointColorItem || ![_availableColors containsObject:_defaultPointColorItem])
     {
-        _defaultPointColorItem = [[OAColorItem alloc] initWithKey:@"purple" value:0xFF3F51B5 isDefault:YES];
+        _defaultPointColorItem = [[OAColorItem alloc] initWithKey:@"default_favorite" value:0xFF3F51B5 isDefault:YES];
```
Comment (PaulStets) [inline Sources/Helpers/Appearance/OAGPXAppearanceCollection.mm:425]: Do not hard-code colors!

File: Sources/Helpers/Appearance/OAGPXAppearanceCollection.mm [hunk 3]
```diff
@@ -461,10 +422,11 @@ - (OAColorItem *)getDefaultPointColorItem
...
         [_availableColors addObject:_defaultPointColorItem];
         _defaultPointColorItem.sortedPosition = [_availableColors indexOfObject:_defaultPointColorItem];
         [_defaultPointColorItem generateId];
+        _defaultColorValues[@"default_favorite"] = @(0xFF3F51B5);
```
Comment (PaulStets) [inline Sources/Helpers/Appearance/OAGPXAppearanceCollection.mm:429]: Do not hard-code colors!

File: Sources/Controllers/TargetMenu/OAEditPointViewController.mm [hunk 4]
```diff
@@ -490,26 +490,8 @@ - (void)createIconList
...
-    else
-    {
-        _selectedColorItem = [_appearanceCollection getColorForGpxFilePath:_waypoint ? _waypoint.docPath : _gpxFileName
-                                                                 groupName:_waypoint ? _waypoint.point.type : @""
-                                                                 pointName:self.name
-                                                              defaultValue:[OAUtilities colorToNumberFromString:[loadedColor toHexARGBString]]];
-    }
-    if (!_selectedColorItem)
-        _selectedColorItem = [_appearanceCollection getDefaultPointColorItem];
+    _selectedColorItem = [_appearanceCollection getColorItemWithValue:[OAUtilities colorToNumberFromString:[[_pointHandler getColor] toHexARGBString]]];
```
Comment (alex-osm) [inline Sources/Controllers/TargetMenu/OAEditPointViewController.mm:494]: Extra double conversion. Introduce `toARGB` method to `UIColor (util)` (OAUtilities)

File: Sources/Helpers/Appearance/OAGPXAppearanceCollection.h [hunk 5]
```diff
@@ -64,17 +61,10 @@
...
+- (void)selectColor:(OAColorItem *)colorItem;
 - (NSArray<OAColorItem *> *)getAvailableColorsSortingByKey;
 - (NSArray<OAColorItem *> *)getAvailableColorsSortingByLastUsed;
-- (void)removeItem:(NSString *)item;
-- (void)removeFavoriteGroupName:(NSString *)groupName pointName:(NSString *)pointName;
-- (void)removeGpxFilePath:(NSString *)gpxFilePath groupName:(NSString *)groupName pointName:(NSString *)pointName;
-- (OAColorItem *)getColorForItem:(NSString *)item defaultValue:(NSInteger)defaultValue;
-- (OAColorItem *)getColorForFavoriteGroupName:(NSString *)groupName pointName:(NSString *)pointName defaultValue:(NSInteger)defaultValue;
-- (OAColorItem *)getColorForGpxFilePath:(NSString *)gpxFilePath groupName:(NSString *)groupName pointName:(NSString *)pointName defaultValue:(NSInteger)defaultValue;
+- (OAColorItem *)getColorItemWithValue:(NSInteger)defaultValue;
```
Comment (alex-osm) [inline Sources/Helpers/Appearance/OAGPXAppearanceCollection.h:67]: `NSInteger` is long, but color value is `int`

**Pull request merged**